uk.co.compucorp.civicrm.tasksassignments/js/test/filters/due_test.js

```diff
@@ -61,14 +108,66 @@ define([
         expect($filter('filterByDue')(mock, 'dateRange', pastDateRange).length).toBe(0);
       });
+
+      it('shows the result for "dateRange" if from and until dates are empty', function() {
+        expect($filter('filterByDue')(mock, 'dateRange', emptyDateRange).length).toBe(1);
+      });
+
+      it('shows the result for "dateRange" if from date is empty and until date is in the future but less than due date', function() {
+        expect($filter('filterByDue')(mockPlusThree, 'dateRange', emptyFromDateAndUntilDateInTheFuture).length).toBe(0);
+      });
+
+      it('shows the result for "dateRange" if from date is empty and until date is in the future but equals due date', function() {
+        expect($filter('filterByDue')(mockPlusTwo, 'dateRange', emptyFromDateAndUntilDateInTheFuture).length).toBe(1);
+      });
+
+      it('shows the result for "dateRange" if from date is empty and until date is in the future but bigger than due date', function() {
+        expect($filter('filterByDue')(mock, 'dateRange', emptyFromDateAndUntilDateInTheFuture).length).toBe(1);
+      });
+
+      it('shows the result for "dateRange" if from date is empty and until date is in the past', function() {
+        expect($filter('filterByDue')(mock, 'dateRange', emptyFromDateAndUntilDateInPast).length).toBe(0);
+      });
+
+      it('shows the result for "dateRange" if from date is empty and until date is today', function() {
+        expect($filter('filterByDue')(mock, 'dateRange', emptyFromDateAndUntilDateIsToday).length).toBe(0);
+      });
+
+      it('shows the result for "dateRange" if until date is empty and from date is in the past', function() {
+        expect($filter('filterByDue')(mock, 'dateRange', emptyUntilDateAndFromDateInThePast).length).toBe(1);
+      });
+
+      it('shows the result for "dateRange" if until date is empty and from date is today', function() {
+        expect($filter('filterByDue')(mock, 'dateRange', emptyUntilDateAndFromDateIsToday).length).toBe(1);
+      });
+
+      it('shows the result for "dateRange" if until date is empty and from date is in the future but less than due date', function() {
+        expect($filter('filterByDue')(mockPlusThree, 'dateRange', emptyUntilDateAndFromDateInTheFuture).length).toBe(1);
+      });
+
+      it('shows the result for "dateRange" if until date is empty and from date is in the future but equals due date', function() {
+        expect($filter('filterByDue')(mockPlusTwo, 'dateRange', emptyUntilDateAndFromDateInTheFuture).length).toBe(1);
+      });
+
+      it('shows the result for "dateRange" if until date is empty and from date is in the future but bigger than due date', function() {
+        expect($filter('filterByDue')(mock, 'dateRange', emptyUntilDateAndFromDateInTheFuture).length).toBe(0);
+      });
     });
 
     describe('due date is in the past', function() {
-      var mock;
+      var mock, mockMinusTwo, mockMinusThree;
```

Comment:
I would suggest `var mockDueDate, mockMinusTwoDays, mockMinusThreeDays;` to make it more readable.  And see we are repeating the `var mock`. So this should be moved to first before each in this test file and replace `var mock` with `var mockDueDate` and change the names in every instance in this test file.